src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -0,0 +1,216 @@
+    <_GenerateResourceCaseMapFile>$(_DesignerIntermediateOutputPath)case_map.txt</_GenerateResourceCaseMapFile>
+  </PropertyGroup>
+  <Message Text="_OuterIntermediateOutputPath: $(_OuterIntermediateOutputPath)" />
+  <Message Text="IntermediateOutputPath: $(IntermediateOutputPath)" />
+</Target>
+
+<Target Name="_GenerateRtxt"
+    Condition="'$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' "
+    DependsOnTargets="_CreatePropertiesCache;_ResolveSdks;_ResolveAndroidTooling;_GetJavaPlatformJar;_GenerateAndroidResourceDir;_SetupDesignerProperties"
+    Inputs="$(_AndroidResFlagFile);@(_AndroidResourceDest);@(LibraryResourceDirectories->'%(StampFile)')"
+    Outputs="$(_DesignerIntermediateOutputPath)R.txt"
+  >
+    <!-- Generate an R.txt file using the Managed Parser -->
+    <GenerateRtxt
+      RTxtFile="$(_DesignerIntermediateOutputPath)R.txt"
+      ResourceDirectory="$(MonoAndroidResDirIntermediate)"
+      AdditionalResourceDirectories="@(LibraryResourceDirectories)"
+      JavaPlatformJarPath="$(JavaPlatformJarPath)"
+      ResourceFlagFile="$(_AndroidResFlagFile)"
+      CaseMapFile="$(_GenerateResourceCaseMapFile)"
+    >
+    </GenerateRtxt>
+</Target>
+
+<Target Name="_GenerateResourceCaseMap"
+    DependsOnTargets="_ComputeAndroidResourcePaths;_SetupDesignerProperties;_GetLibraryImports"
+    Inputs="@(_AndroidResourceDest);@(LibraryResourceDirectories->'%(StampFile)')"
+    Outputs="$(_GenerateResourceCaseMapFile)"
+  >
+    <!-- Generate a ResourceMap file for the project and its resources -->
+    <GenerateResourceCaseMap
+      Resources="@(_AndroidResourceDest)"
+      AdditionalResourceDirectories="@(LibraryResourceDirectories)"
+      ProjectDir="$(ProjectDir)"
+      ResourceDirectory="$(MonoAndroidResDirIntermediate)"
+      OutputFile="$(_GenerateResourceCaseMapFile)"
+    />
+    <Touch Files="$(_GenerateResourceCaseMapFile)" />
+    <ItemGroup>
+      <FileWrites Include="$(_GenerateResourceCaseMapFile)" />
+    </ItemGroup>
+</Target>
+
+<!--
+    We only want to do this for non application projects unless the user has
+    specifically opted into the new system. This can be done by either
+    1. using `dotnet new android` and building a new project.
+    2. Add <AndroidUseDesignerAssembly>true</AndroidUseDesignerAssembly> to
+       the project.
+ -->
+<Target Name="_GenerateResourceDesignerIntermediateClass"
+    Inputs="$(MSBuildProjectFullPath)"
+    Outputs="$(_GenerateResourceDesignerClassFile)"
+    Condition=" '$(AndroidUseDesignerAssembly)' == 'True' "
+  >
+  <!-- Creates a Resource.designer.cs file in the Intermediate output path which derives from the
+       Designer Assembly, for backward compatability.
+   -->
+   <GenerateResourceDesignerIntermediateClass
+      Namespace="$(AndroidResgenNamespace)"
```

Comment:
It's odd that here you indent *three* spaces, not 4 (desired), not 2 (done elsewhere).